trunner/config.py

```diff
@@ -157,16 +157,25 @@ def _parse_shell_command(self, config: dict):
             raise ParserError("execute/run attribute cannot be empty")
 
         if execute_binary:
-            prefix = self.ctx.target.exec_dir() + "/" if self.ctx.target.rootfs else "sysexec "
-            parsed_cmd = shlex.split(prefix + cmd[0]) + cmd[1:]
-            binary = cmd[0]
+            path_cmd = Path(cmd[0])
+            prefix = (
+                ("" if path_cmd.is_absolute() else f"{self.ctx.target.exec_dir()}/")
+                if self.ctx.target.rootfs else "sysexec "
+            )
+
+            binary = str(path_cmd if self.ctx.target.rootfs else path_cmd.name)
+            parsed_cmd = shlex.split(prefix + binary) + cmd[1:]
+
+            path = path_cmd.parent if path_cmd.is_absolute() else self.ctx.target.exec_dir()
         else:
             binary = None
             parsed_cmd = cmd
+            path = Path()
 
         self.test.shell = ShellOptions(
             binary=binary,
             cmd=parsed_cmd,
+            path=path.relative_to("/"),
```

Comment:
As I suggested before it wuld be good to place a comment why it's done

```suggestion
            # when launching host tools like phoenixd absolute path from phoenix has to be combined with absolute path for host.
            # Two absolute paths can't be combined, so phoenix path is transformed to a relative one
            path=path.relative_to("/"),
```